vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -427,34 +431,24 @@ export class WebRequestWorkerSingleton
      */
     protected async makeWebRequest(url: string, ctx: IAcquisitionWorkerContext, throwOnError: boolean, numRetries: number): Promise<string | undefined>
     {
+        ctx.eventStream.post(new WebRequestInitiated(`Making Web Request For ${url}`));
         const options = await this.getAxiosOptions(ctx, numRetries);
 
         try
         {
             ctx.eventStream.post(new WebRequestSent(url));
-            const response = await this.axiosGet(url, ctx, { transformResponse: (x: any) => x as any, ...options }
-            );
-
-            if (response?.headers?.['content-type'] === 'application/json')
-            {
-                try
-                {
-                    // Try to copy logic from https://github.com/axios/axios/blob/2e58825bc7773247ca5d8c2cae2ee041d38a0bb5/lib/defaults/index.js#L100
-                    const jsonData = JSON.parse(response?.data ?? null);
-                    if (jsonData) // JSON.parse(null) => null but JSON.parse(undefined) => SyntaxError. We only want to return undefined and not null based on funct signature.
-                    {
-                        return jsonData;
-                    };
-                }
-                catch (error: any)
-                {
-
-                }
-            }
-            return response?.data;
+            const response = await this.axiosGet(url, ctx, { ...options });
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            return this.client !== null ? response?.data : response;
```

Comment:
Why do we care if the client is null here? I might've expected more like `return response?.data ?? response;` If we do care about client being null, can we skip the previous line?

Reply:
Hm, I think the above code is misleading since axiosGet will call fetch instead of axios if client is null. I should rename the function so that it's more clear and possibly make it always output the same data format.

Reply:
I've improved this!